### PR TITLE
fixed potentially inconsistent states

### LINKED ISSUE #119: Fix potentially inconsistent state in plan
### Expected Behavior

Use callback form for all `setState()`s in `Plan.tsx`

### Current Behavior

[State is updated with potentially stale values](https://lgtm.com/projects/g/drklee3/YOUtinerary/latest/files/packages/web/src/components/Plan.tsx?sort=name&dir=ASC&mode=heatmap)

More info: https://lgtm.com/rules/1819283066/

### PR DESCRIPTION
**Pull request by za-wong**
### Description

made changes to all setState's in plan.tsx

followed documentation on fixing potentially inconsistent states

### Tested

yarn run dev and nothing broke lol

### Related issues

* Fixes #119

### CODE REVIEW

**Comment by za-wong** [comment]
very loose test.

File: packages/web/src/components/Plan.tsx [hunk 1]
```diff
@@ -181,50 +181,60 @@ export class Plan extends React.Component<{}, State> {
...
+        this.setState((prevState) => ({
+            events: editEvent(prevState.events, modified),
+        }));
     };
 
     editEventName = (id: number, name: string): void => {
-        this.setState({ events: editEventName(this.state.events, id, name) });
+        this.setState((prevState) => {
+            events: editEventName(prevState.events, id, name);
+        });
```
Comment (drklee3) [inline packages/web/src/components/Plan.tsx]: Missing parenthesis around the `{events: ...}`

**Review by drklee3 — CHANGES_REQUESTED** [review]
Looks good, just 1 small change

**Comment by za-wong** [comment]
for all of them, or just that function in particular?

**Comment by drklee3** [comment]
> for all of them, or just that function in particular?

Only one of them was missing it, so just that one.

**Review by drklee3 — APPROVED** [review]
lgtm!

**Pull request merged**